main/src/vocab/nllb_vocab.rs

```diff
@@ -0,0 +1,245 @@
+    "taq_Latn", "taq_Tfng", "tpi_Latn", "tsn_Latn", "tso_Latn", "tuk_Latn", "tum_Latn", "tur_Latn",
+    "twi_Latn", "tzm_Tfng", "uig_Arab", "ukr_Cyrl", "umb_Latn", "urd_Arab", "uzn_Latn", "vec_Latn",
+    "vie_Latn", "war_Latn", "wol_Latn", "xho_Latn", "ydd_Hebr", "yor_Latn", "yue_Hant", "zho_Hans",
+    "zho_Hant", "zul_Latn",
+];
+
+pub struct NLLBVocab {
+    /// A mapping of tokens as string to indices (i.e. the encoder base)
+    pub values: HashMap<String, i64>,
+
+    /// A mapping of token IDs to strings (i.e. the decoder base)
+    pub indices: HashMap<i64, String>,
+
+    /// A mapping of special value tokens as strings to IDs (i.e. the encoder base for special
+    /// values), special values typically include things like BOS/EOS markers, class markers, mask
+    /// markers and padding markers
+    pub special_values: HashMap<String, i64>,
+
+    /// A mapping of special value tokens as IDs to strings (i.e. the decoder base for special values)
+    pub special_indices: HashMap<i64, String>,
+
+    /// Language code stored as bytes for extraction of the prefix in input sequences
+    pub language_codes_bytes: HashSet<Vec<u8>>,
+
+    pub special_token_storage: SpecialTokenMap,
+}
+
+impl NLLBVocab {
+    /// The beginning of sequence token that was used during pretraining.
+    /// Can be used a sequence classifier token.
+    pub fn get_bos_value(&self) -> &str {
+        self.special_token_storage
+            .bos_token
+            .as_deref()
+            .unwrap_or("<s>")
+    }
+
+    /// End of sequence token.
+    pub fn get_eos_value(&self) -> &str {
+        self.special_token_storage
+            .eos_token
+            .as_deref()
+            .unwrap_or("</s>")
+    }
+
+    /// Returns the SEP token for M2M100 (`</s>`)
+    pub fn get_sep_value(&self) -> &str {
+        self.special_token_storage
+            .eos_token
+            .as_deref()
+            .unwrap_or("</s>")
+    }
+
+    /// Returns the PAD token for M2M100 (`<pad>`)
+    pub fn get_pad_value(&self) -> &str {
+        self.special_token_storage
+            .pad_token
+            .as_deref()
+            .unwrap_or("<pad>")
+    }
+}
+
+impl Vocab for NLLBVocab {
+    fn get_unknown_value(&self) -> &str {
+        self.special_token_storage.unk_token.as_str()
+    }
+
+    fn values(&self) -> &HashMap<String, i64> {
+        &self.values
+    }
+
+    fn indices(&self) -> &HashMap<i64, String> {
+        &self.indices
+    }
+
+    fn special_values(&self) -> &HashMap<String, i64> {
+        &self.special_values
+    }
+
+    fn special_indices(&self) -> &HashMap<i64, String> {
+        &self.special_indices
+    }
+
+    fn from_file<V: AsRef<Path>>(vocab: V) -> Result<Self, TokenizerError> {
+        let tokenizer = Tokenizer::deserialize(vocab)?;
+        let special_config = SpecialTokenMap {
+            unk_token: "<unk>".into(),
+            ..Default::default()
```

Comment:
This is dangerous as it would change the behaviour of the vocab if the shared `SpecialTokenMap` default changes for some reason. Can you please initialize as in https://github.com/guillaume-be/rust-tokenizers/blob/4d8aa3eb964fd33cb0dbc6bfd86b9edb5d264faf/main/src/vocab/m2m100_vocab.rs#L124

Reply:
Fixed in https://github.com/npatsakula/rust-tokenizers/pull/5.